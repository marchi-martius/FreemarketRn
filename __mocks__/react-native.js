const rn = require('react-native');

jest.mock('Linking', () => {
  return {
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    openURL: jest.fn(),
    canOpenURL: jest.fn(),
    getInitialURL: jest.fn(),
  };
});

jest.mock('../app/api/firebase', () => {
  return {
    saveImageToStorage: jest.fn(),
  }
});

module.exports = rn;
