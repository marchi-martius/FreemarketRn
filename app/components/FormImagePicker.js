import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { View, Image, Text, Button, StyleSheet } from 'react-native';

import { selectPhotoTapped } from '../services/ImagePickerService';

const styles = StyleSheet.create({
  formImage: {
    alignItems: 'center',
  },
  imageSection: {
    width: 150,
    height: 150,
  },
  image: {
    flex: 1,
  },
  imagePlaceholder: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    borderWidth: 4,
    borderStyle: 'dotted',
    borderRadius: 1,
  },
  button: {
    paddingTop: 10,
    paddingBottom: 10,
    width: 150,
  },
});

const renderPlaceholder = () => (
  <View style={styles.imagePlaceholder}>
    <Text>Place for an Image</Text>
  </View>
);

export default class FormImagePicker extends Component {
  selectPhoto() {
    selectPhotoTapped().then((localImage) => {
      this.props.setLocalImage(localImage);
    });
  }

  renderImage() {
    return (
      <Image
        source={{ uri: this.props.imageSource }}
        style={styles.image}
      />
    );
  }

  renderImageBlock() {
    if (this.props.imageSource) {
      return this.renderImage();
    } else {
      return renderPlaceholder();
    }
  }

  render() {
    return (
      <View style={styles.formImage}>
        <View style={styles.imageSection}>
          { this.renderImageBlock() }
        </View>
        <View style={styles.button}>
          <Button
            title="Select Image"
            onPress={() => this.selectPhoto()}
          />
        </View>
      </View>
    );
  }
}

FormImagePicker.propTypes = {
  imageSource: PropTypes.string,
  setLocalImage: PropTypes.func.isRequired,
};

FormImagePicker.defaultProps = {
  imageSource: null,
};
