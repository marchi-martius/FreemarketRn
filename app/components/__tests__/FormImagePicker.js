import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import FormImagePicker from '../FormImagePicker';

it('renders successfully', () => {
  const tree = renderer.create(
    <FormImagePicker
      imageSource={''}
      setLocalImage={() => true}
    />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
