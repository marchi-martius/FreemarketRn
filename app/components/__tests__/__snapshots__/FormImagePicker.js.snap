// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders successfully 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <View
    style={
      Object {
        "height": 150,
        "width": 150,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 1,
          "borderStyle": "dotted",
          "borderWidth": 4,
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Place for an Image
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "paddingBottom": 10,
        "paddingTop": 10,
        "width": 150,
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0C42FD",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Select Image
        </Text>
      </View>
    </View>
  </View>
</View>
`;
